### ISSUE TITLE
https://karpenter.sh/v0.27.3/concepts/deprovisioning/#consolidation

### ISSUE THREAD

**Issue by imagekitio** [aws/karpenter#3894]
### What karpenter features are relevant?

Consolidation

### How should the docs be improved?

The [consolidation docs](https://karpenter.sh/v0.27.3/concepts/deprovisioning/#consolidation) says

>For spot nodes, Karpenter only uses the deletion consolidation mechanism. It will not replace a spot node with a cheaper spot node. Spot instance types are selected with the `price-capacity-optimized` strategy and often the cheapest spot instance type is not launched due to the likelihood of interruption. Consolidation would then replace the spot instance with a cheaper instance negating the `price-capacity-optimized` strategy entirely and increasing interruption rate.

Its fine to not replace a spot instance of same capacity (cpu/memory) with a cheaper spot instance of similar capacity. However it is not clear from the documentation, what would happen if a bigger instance is not being fully utilise. Will it be replaced with a smaller instance.

For example, let's say I have configured a provisioner with following instance types with `spot` capacity only. If let's say `c6i.32xlarge` is running and only being utilised 25% (i.e. 32 cores), will consolidation launch a 32-core machine and shutdown `c6i.32xlarge`.

Instance Info | vCPU | Memory GiB
-- | -- | --
c5.4xlarge | 16 | 32
c6i.8xlarge | 32 | 64
r5.8xlarge | 32 | 256
r6i.8xlarge | 32 | 256
c6i.12xlarge | 48 | 96
r5.12xlarge | 48 | 384
m5.12xlarge | 48 | 192
c6i.16xlarge | 64 | 128
m5d.24xlarge | 96 | 384
r5d.metal | 96 | 768
c6i.metal | 128 | 256
c6i.32xlarge | 128 | 256

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by imagekitio** [comment]
Thanks. 

Can you please confirm if this is correct understanding in above scenario. Setting `ttlSecondsUntilExpired` will cause Karpenter to request replacement nodes and that should launch a smaller instance if running pods on to-be-deleted node requires less capacity?

**Comment by tzneal** [comment]
Yes, if the node expires Karpenter will re-do the scheduling calculation and potentially launch a smaller node.
